### PR TITLE
Chapter2/homework/1

### PR DESCRIPTION
**Pull request by chatieb**
Chapter 2 Homework 1: OOP 
Implemented the Chapter 1 Homework 8 using OOP style.  Also, required code changes, because the corrupted file changed data since Homework 8 was given out, such as the dollar sign was changed from '€' to '£'.

### CODE REVIEW

File: Src/BootCamp.Chapter/BalanceStats.cs [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+
+            if (lastComma > 0)
+            {
+                name = name.Remove(lastComma, 2).Insert(lastComma, " and ");
+            }
+
+            return name;
+        }
+
+        private bool IsArrayNullOrEmpty(string[] array)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/BalanceStats.cs]: Codacy Issue found: [Make 'IsArrayNullOrEmpty' a static method.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5063788)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
- Added 25
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/BalanceStats.cs  6
- Src/BootCamp.Chapter/TextTable.cs  7
- Src/BootCamp.Chapter/InvalidBalancesException.cs  1
- Src/BootCamp.Chapter/CleanTable.cs  8
         

Clones added
============
- Src/BootCamp.Chapter/BalanceStats.cs  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5063788&bid=16663551)

File: Src/BootCamp.Chapter/CleanTable.cs [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+                return true;
+            }
+            return false;
+        }
+
+        private bool IsValidBalance(string balance)
+        {
+            for (var i = 0; i < balance.Length; i++)
+            {
+                if (balance[i] != DollarSign && (balance[i] < '0' || balance[i] > '9') && balance[i] != '-' && balance[i] != '.')
```
Comment (Almantask) [inline Src/BootCamp.Chapter/CleanTable.cs]: Codacy Issue found: [Reduce the number of conditional operators (4) used in the expression (maximum allowed 3).](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5063788)

File: Src/BootCamp.Chapter/BalanceStats.cs [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+                {
+                    mostPoorPerson.Append(", ");
+                }
+
+                mostPoorPerson.Append(peopleBalances[0]);
+                minBalance = balance;
+            }
+        }
+
+        private string FixPlural(string name)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/BalanceStats.cs]: Codacy Issue found: [Make 'FixPlural' a static method.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5063788)

File: Src/BootCamp.Chapter/CleanTable.cs [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        private bool IsValidLetter(char letter)
+        {
+            if ((letter >= 'a' && letter <= 'z') ||
```
Comment (Almantask) [inline Src/BootCamp.Chapter/CleanTable.cs]: Codacy Issue found: [Reduce the number of conditional operators (6) used in the expression (maximum allowed 3).](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5063788)

File: Src/BootCamp.Chapter/TextTable.cs [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+
+            if (endPos <= message.Length)
+            {
+                messageLine[messageCount] = message.Substring(beginPos, message.Length - beginPos);
+            }
+
+            return messageLine;
+        }
+
+        private int GetMaxMessageLineLength(string[] messageLine)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/TextTable.cs]: Codacy Issue found: [Make 'GetMaxMessageLineLength' a static method.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5063788)

File: Src/BootCamp.Chapter/BalanceStats.cs [hunk 6]
```diff
@@ -0,0 +1,258 @@
...
+            }
+
+            CultureInfo.CurrentCulture = CultureInfo.InvariantCulture;
+
+            var minBalanceNoComma = RemoveComma($"{minBalance:C0}").Replace("(", "-").Replace(")", "");
+            var word = name.ToString().Contains(", ") ? "have" : "has";
+            return $"{FixPlural(name.ToString())} {word} the least money. {minBalanceNoComma}.";
+        }
+
+        private void GetPoorest(ref StringBuilder mostPoorPerson, ref decimal minBalance, string[] peopleBalances, decimal balance)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/BalanceStats.cs]: Codacy Issue found: [Make 'GetPoorest' a static method.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5063788)

File: Src/BootCamp.Chapter/CleanTable.cs [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+                (letter == ' ') ||
+                (letter == '\'') ||
+                (letter == '-'))
+            {
+                return true;
+            }
+            return false;
+        }
+
+        private bool HasPeriodAtEnd(string name)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/CleanTable.cs]: Codacy Issue found: [Make 'HasPeriodAtEnd' a static method.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5063788)

File: Src/BootCamp.Chapter/TextTable.cs [hunk 8]
```diff
@@ -0,0 +1,173 @@
...
+                table.Append(tableLine + Environment.NewLine);
+            }
+
+            tableLine = BuildBorderLine(maxLength, padding, "+", "-");
+            table.Append(tableLine + Environment.NewLine);
+
+            return table.ToString();
+        }
+
+        private string BuildBorderLine(int maxLength, int padding, string sideBorder, string edgeBorder)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/TextTable.cs]: Codacy Issue found: [Make 'BuildBorderLine' a static method.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5063788)

File: Src/BootCamp.Chapter/TextTable.cs [hunk 9]
```diff
@@ -0,0 +1,173 @@
...
+                if (message[i] == '\r' && message[i + 1] == '\n')
+                {
+                    numOfLines++;
+                }
+            }
+
+            return numOfLines;
+        }
+
+        private string[] GetMessageLines(string message, int numOfMessageLines)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/TextTable.cs]: Codacy Issue found: [Make 'GetMessageLines' a static method.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5063788)

File: Src/BootCamp.Chapter/CleanTable.cs [hunk 10]
```diff
@@ -0,0 +1,171 @@
+﻿using System;
+using System.Collections.Generic;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/CleanTable.cs]: Codacy Issue found: [Remove this unnecessary 'using'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5063788)

File: Src/BootCamp.Chapter/CleanTable.cs [hunk 11]
```diff
@@ -0,0 +1,171 @@
...
+            var buildLines = new StringBuilder();
+            var skipCarriageReturn = corruptedBalanceLines.Length - 1;
+
+            for (var i = 0; i < corruptedBalanceLines.Length; i++)
+            {
+                fixedBalanceLines[i] = corruptedBalanceLines[i].Replace("_", "");
+
+                if (!isValidLine(fixedBalanceLines[i]))
+                {
+                    throw new Exception("Invalid Balance.");
```
Comment (Almantask) [inline Src/BootCamp.Chapter/CleanTable.cs]: Codacy Issue found: ['System.Exception' should not be thrown by user code.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5063788)

File: Src/BootCamp.Chapter/CleanTable.cs [hunk 12]
```diff
@@ -0,0 +1,171 @@
...
+using System.IO;
+using System.Text;
+
+namespace BootCamp.Chapter
+{
+    class CleanTable
+    {
+        private const char DollarSign = '£';
+        private string _corruptedBalanceFile;
+        private string _fixedBalanceFile;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/CleanTable.cs]: Codacy Issue found: [Make '_fixedBalanceFile' 'readonly'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5063788)

File: Src/BootCamp.Chapter/TextTable.cs [hunk 13]
```diff
@@ -0,0 +1,173 @@
...
+            }
+            var numOfMessageLines = GetNumberOfMessageLines(message);
+            var messageLine = GetMessageLines(message, numOfMessageLines);
+            var maxLength = GetMaxMessageLineLength(messageLine);
+
+            //Build table from message
+            return BuildTable(messageLine, maxLength, padding);
+        }
+
+        private int GetNumberOfMessageLines(string message)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/TextTable.cs]: Codacy Issue found: [Make 'GetNumberOfMessageLines' a static method.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5063788)

File: Src/BootCamp.Chapter/CleanTable.cs [hunk 14]
```diff
@@ -0,0 +1,171 @@
...
+            var fixedBalanceLines = new string[corruptedBalanceLines.Length];
+            FixCorruptedBalanceFile(corruptedBalanceLines, fixedBalanceLines);
+
+            DisplayStasticalData(CreateBalanceLinesForTextTable(fixedBalanceLines));
+        }
+
+        private string[] CreateBalanceLinesForTextTable(string[] fixedBalanceLines)
+        {
+            var balanceLinesForTextTable = new string[fixedBalanceLines.Length];
+            var balanceSeperator = ",";
```
Comment (Almantask) [inline Src/BootCamp.Chapter/CleanTable.cs]: Codacy Issue found: [Add the 'const' modifier to 'balanceSeperator'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5063788)

File: Src/BootCamp.Chapter/TextTable.cs [hunk 15]
```diff
@@ -0,0 +1,173 @@
...
+            for (var i = 0; i < maxLength + 2 * padding; i++)
+            {
+                line.Append(edgeBorder);
+            }
+            line.Append(sideBorder);
+
+            return line.ToString();
+        }
+
+        private string BuildMessageLine(int maxLength, int padding, string sideBorder, string messageLine)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/TextTable.cs]: Codacy Issue found: [Make 'BuildMessageLine' a static method.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5063788)

File: Src/BootCamp.Chapter/BalanceStats.cs [hunk 16]
```diff
@@ -0,0 +1,258 @@
...
+                    {
+                        biggestLoss = GetBiggestLoss(biggestLoss, beginAmount, endAmount);
+                    }
+                }
+            }
+
+            return biggestLoss;
+        }
+
+        private decimal GetBiggestLoss(decimal biggestLoss, decimal beginAmount, decimal endAmount)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/BalanceStats.cs]: Codacy Issue found: [Make 'GetBiggestLoss' a static method.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5063788)

File: Src/BootCamp.Chapter/CleanTable.cs [hunk 17]
```diff
@@ -0,0 +1,171 @@
...
+        public void Clean()
+        {
+            var corruptedBalanceLines = File.ReadAllLines(_corruptedBalanceFile);
+            var fixedBalanceLines = new string[corruptedBalanceLines.Length];
+            FixCorruptedBalanceFile(corruptedBalanceLines, fixedBalanceLines);
+
+            DisplayStasticalData(CreateBalanceLinesForTextTable(fixedBalanceLines));
+        }
+
+        private string[] CreateBalanceLinesForTextTable(string[] fixedBalanceLines)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/CleanTable.cs]: Codacy Issue found: [Make 'CreateBalanceLinesForTextTable' a static method.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5063788)

File: Src/BootCamp.Chapter/CleanTable.cs [hunk 18]
```diff
@@ -0,0 +1,171 @@
...
+        private bool HasPeriodAtEnd(string name)
+        {
+            if (name[name.Length - 1] == '.')
+            {
+                return true;
+            }
+            return false;
+        }
+
+        private bool IsValidBalance(string balance)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/CleanTable.cs]: Codacy Issue found: [Make 'IsValidBalance' a static method.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5063788)

File: Src/BootCamp.Chapter/CleanTable.cs [hunk 19]
```diff
@@ -0,0 +1,171 @@
...
+                if (!IsValidLetter(name[i]) && !HasPeriodAtEnd(name))
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        private bool IsValidLetter(char letter)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/CleanTable.cs]: Codacy Issue found: [Make 'IsValidLetter' a static method.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5063788)

File: Src/BootCamp.Chapter/BalanceStats.cs [hunk 20]
```diff
@@ -0,0 +1,258 @@
...
+                SetNames(ref name, ref maxBalance, peopleBalances, balance);
+            }
+
+            CultureInfo.CurrentCulture = CultureInfo.InvariantCulture;
+
+            var maxBalanceNoComma = RemoveComma($"{maxBalance:C0}");
+            return $"{FixPlural(name.ToString())} had the most money ever. {maxBalanceNoComma}.";
+        }
+
+        private decimal GetLargestBalance(string[] peopleBalances)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/BalanceStats.cs]: Codacy Issue found: [Make 'GetLargestBalance' a static method.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5063788)

File: Src/BootCamp.Chapter/CleanTable.cs [hunk 21]
```diff
@@ -0,0 +1,171 @@
...
+                if (i < skipCarriageReturn)
+                {
+                    buildLines.Append("\r\n");
+                }
+            }
+
+            File.WriteAllText(_fixedBalanceFile, buildLines.ToString());
+        }
+
+        private bool isValidLine(string line)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/CleanTable.cs]: Codacy Issue found: [Rename method 'isValidLine' to match pascal case naming rules, consider using 'IsValidLine'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5063788)

File: Src/BootCamp.Chapter/BalanceStats.cs [hunk 22]
```diff
@@ -0,0 +1,258 @@
...
+                if (isBalanceValid && balance > maxBalance)
+                {
+                    maxBalance = balance;
+                }
+            }
+
+            return maxBalance;
+        }
+
+        private void SetNames(ref StringBuilder name, ref decimal maxBalance, string[] peopleBalances, decimal balance)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/BalanceStats.cs]: Codacy Issue found: [Make 'SetNames' a static method.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5063788)

File: Src/BootCamp.Chapter/CleanTable.cs [hunk 23]
```diff
@@ -0,0 +1,171 @@
...
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+
+namespace BootCamp.Chapter
+{
+    class CleanTable
+    {
+        private const char DollarSign = '£';
+        private string _corruptedBalanceFile;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/CleanTable.cs]: Codacy Issue found: [Make '_corruptedBalanceFile' 'readonly'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5063788)

File: Src/BootCamp.Chapter/CleanTable.cs [hunk 24]
```diff
@@ -0,0 +1,171 @@
...
+                if (balance[i] != DollarSign && (balance[i] < '0' || balance[i] > '9') && balance[i] != '-' && balance[i] != '.')
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        private void DisplayStasticalData(string[] peopleBalances)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/CleanTable.cs]: Codacy Issue found: [Make 'DisplayStasticalData' a static method.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5063788)

File: Src/BootCamp.Chapter/BalanceStats.cs [hunk 25]
```diff
@@ -0,0 +1,258 @@
...
+        {
+            if (array == null || array.Length == 0)
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        private string RemoveComma(string data)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/BalanceStats.cs]: Codacy Issue found: [Make 'RemoveComma' a static method.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5063788)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
- Added 6
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/TextTable.cs  7
- Src/BootCamp.Chapter/BalanceStats.cs  6
- Src/BootCamp.Chapter/CleanTable.cs  5
- Src/BootCamp.Chapter/InvalidBalancesException.cs  1
         

Clones added
============
- Src/BootCamp.Chapter/BalanceStats.cs  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5063788&bid=16663551)

File: Src/BootCamp.Chapter/BalanceStats.cs [hunk 26]
```diff
@@ -0,0 +1,258 @@
...
+            var maxBalanceNoComma = RemoveComma($"{maxBalance:C0}");
+            var word1 = name.ToString().Contains(", ") ? "are" : "is";
+            var word2 = name.ToString().Contains(", ") ? "people" : "person";
+            return $"{FixPlural(name.ToString())} {word1} the richest {word2}. {maxBalanceNoComma}.";
+        }
+
+        /// <summary>
+        /// Return name and current money of the most poor person.
+        /// </summary>
+        public string FindMostPoorPerson(string[] peopleAndBalances)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/BalanceStats.cs]: Codacy Issue found: [Make 'FindMostPoorPerson' a static method.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5063788)

File: Src/BootCamp.Chapter/BalanceStats.cs [hunk 27]
```diff
@@ -0,0 +1,258 @@
...
+                return NotAvailable;
+            }
+
+            CultureInfo.CurrentCulture = CultureInfo.InvariantCulture;
+
+            var biggestLossEverNoComma = RemoveComma($"{biggestLossEver:C0}").Replace("(", "-").Replace(")", "");
+            return $"{name} lost the most money. {biggestLossEverNoComma}.";
+        }
+
+        private decimal GetBiggestLossForPerson(string[] peopleBalances)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/BalanceStats.cs]: Codacy Issue found: [Make 'GetBiggestLossForPerson' a static method.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5063788)

File: Src/BootCamp.Chapter/CleanTable.cs [hunk 28]
```diff
@@ -0,0 +1,202 @@
...
+        private static bool IsValidSymbolForBalance(char punctuation)
+        {
+            if (punctuation == '-' || punctuation == '.')
+            {
+                return true;
+            }
+            return false;
+        }
+
+        private void DisplayStasticalData(string[] peopleBalances)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/CleanTable.cs]: Codacy Issue found: [Make 'DisplayStasticalData' a static method.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5063788)

File: Src/BootCamp.Chapter/BalanceStats.cs [hunk 29]
```diff
@@ -0,0 +1,258 @@
...
+namespace BootCamp.Chapter
+{
+    public class BalanceStats
+    {
+        private const string NotAvailable = "N/A.";
+
+        /// <summary>
+        /// Return name and balance(current) of person who had the biggest historic balance.
+        /// </summary>
+        public string FindHighestBalanceEver(string[] peopleAndBalances)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/BalanceStats.cs]: Codacy Issue found: [Make 'FindHighestBalanceEver' a static method.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5063788)

File: Src/BootCamp.Chapter/BalanceStats.cs [hunk 30]
```diff
@@ -0,0 +1,258 @@
...
+                bigLoss = loss;
+            }
+
+            return bigLoss;
+        }
+
+        /// <summary>
+        /// Return name and current money of the richest person.
+        /// </summary>
+        public string FindRichestPerson(string[] peopleAndBalances)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/BalanceStats.cs]: Codacy Issue found: [Make 'FindRichestPerson' a static method.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5063788)

File: Src/BootCamp.Chapter/TextTable.cs [hunk 31]
```diff
@@ -0,0 +1,173 @@
...
+           +-------+
+
+         */
+
+        /// <summary>
+        /// Build a table for given message with given padding.
+        /// Padding means how many spaces will a message be wrapped with.
+        /// Table itself is made of: "+-" symbolmessage. 
+        /// </summary>
+        public string Build(string message, int padding)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/TextTable.cs]: Codacy Issue found: [Make 'Build' a static method.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5063788)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/CleanTable.cs  5
- Src/BootCamp.Chapter/BalanceStats.cs  6
- Src/BootCamp.Chapter/InvalidBalancesException.cs  1
- Src/BootCamp.Chapter/TextTable.cs  7
         

Clones added
============
- Src/BootCamp.Chapter/BalanceStats.cs  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5063788&bid=16663551)

**Comment by Almantask** [comment]
Please redo the whole thing. Your task is to use OOP, therefrore- don't work with string, work with objects:
- Parse everything into a Person object who has string Name, decimal[] Balances. 
- From a person class, have a method to get current balance (last)
- From a person class have methods for getting highest and smallest balances 
- Have another class, called BalancesStatistics which will process multiple people, using their own methods and give back a certain person or people with highest, lowest and other kinds of balances as requested.

**Comment by chatieb** [comment]
I have re-submitted the assignment with PR #203, because I could not upload it and had to create a new branch and new pull request.

**Pull request closed without merging**